main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -184,6 +184,7 @@ object Parser2 {
             // token.
             SourceLocation(
               isReal = true,
+              s.src,
```

Comment:
I am not sure about this, but shouldn't this be `lastAdvance.src`, or does all parsing for a given state `s` happen inside the same source / file?